apollo-router/src/protocols/multipart.rs

```diff
@@ -104,9 +104,17 @@ impl Stream for Multipart {
 
                     Poll::Ready(Some(Ok(buf)))
                 }
-                Some(MessageKind::Message(mut response)) => {
-                    let is_still_open =
-                        response.has_next.unwrap_or(false) || response.subscribed.unwrap_or(false);
+                Some(MessageKind::Message(response)) => {
+                    let is_still_open = response
+                        .as_object()
+                        .and_then(|o| o.get("hasNext").and_then(|v| v.as_bool()))
+                        .unwrap_or(false)
+                        || response
+                            .as_object()
+                            .and_then(|o| o.get("payload"))
+                            .and_then(|v| v.as_object())
+                            .map(|o| !o.is_empty())
+                            .unwrap_or(false);
```

Comment:
I think it won't work, it's more complicated than this IIRC

Reply:
do you see cases where it would fail, that I could add to unit tests?